Sources/CacheAPI-smf.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+/**
+ * Simple Machines Forum (SMF)
+ *
+ * @package SMF
+ * @author Simple Machines http://www.simplemachines.org
+ * @copyright 2016 Simple Machines and individual contributors
+ * @license http://www.simplemachines.org/about/smf/license.php BSD
+ *
+ * @version 2.1 Beta 3
+ */
+
+if (!defined('SMF'))
+	die('Hacking attempt...');
+
+/**
+ * Our Cache API class
+ * @package cacheAPI
+ */
+class smf_cache extends cache_api
+{
+	/**
+	 * Checks whether we can use the cache method performed by this API.
+	 *
+	 * @access public
+	 * @param boolean $test Test if this is supported or enabled.
+	 * @return boolean Whether or not the cache is supported
+	 */
+	public function isSupported($test = false)
+	{
+		global $cachedir;
+
+		$supported = is_writable($cachedir);
+
+		if ($test)
+			return $supported;
+		return parent::isSupported() && $supported;
+	}
+
+	/**
+	 * Connects to the cache method. If this fails, we return false, otherwise we return true.
+	 *
+	 * @access public
+	 * @return boolean Whether or not the cache method was connected to.
+	 */
+	public function connect()
+	{
+//		$this->setPrefix('');
+
+		// No need to do anything here.
+		return true;
+	}
+
+	/**
+	 * Gets data from the cache.
+	 *
+	 * @access public
+	 * @param string $key The key to use, the prefix is applied to the key name.
+	 * @param string $ttl Overrides the default TTL.
+	 * @return mixed The result from the cache, if there is no data or it is invalid, we return null.
+	 */
+	public function getData($key, $ttl = null)
+	{
+		global $cachedir;
+
+		$key = $this->prefix . strtr($key, ':/', '-_');
+
+		// SMF Data returns $value and $expired.  $expired has a unix timestamp of when this expires.
+		if (file_exists($cachedir . '/data_' . $key . '.php') && filesize($cachedir . '/data_' . $key . '.php') > 10)
+		{
+			// Work around Zend's opcode caching (PHP 5.5+), they would cache older files for a couple of seconds
+			// causing newer files to take effect a while later.
+			if (function_exists('opcache_invalidate'))
+				opcache_invalidate($cachedir . '/data_' . $key . '.php', true);
+
+			if (function_exists('apc_delete_file'))
+				@apc_delete_file($cachedir . '/data_' . $key . '.php');
+
+			// php will cache file_exists et all, we can't 100% depend on its results so proceed with caution
+			@include($cachedir . '/data_' . $key . '.php');
```

Comment:
Could we move to JSON here? IMO it would be better than abusing PHP files for this...